martian/src/metadata.rs

```diff
@@ -136,7 +143,7 @@ fn ignore_not_found(err: std::io::Error) -> std::io::Result<()> {
 }
 
 impl Metadata {
-    pub fn new(mut args: Vec<String>) -> Metadata {
+    pub fn new(mut args: Vec<String>) -> Result<Metadata> {
```

Comment:
This is a breaking API change that should perhaps be called out specifically in the commit message.

Reply:
I don't even think we should make the constructor public. If anything this should be `pub(crate)` too.